Integration with lockitron

aarya123 opened the pull request:
## Description:
Add support to lockitrons that have the bridge extension that enables wi-fi support. Currently only supports one lock at a time.

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml
lock:
  - platform: lockitron
    access_token: asdf
    id: fdsa
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

deftdawg committed:
Improvements for WeMo Insight switches

* Changes current power units to watts
* Adds power on times and additional totals

deftdawg committed:
Update wemo.py

* Reordered datetime import
* Spaces by 4

deftdawg committed:
Update wemo.py

* continuation line under-indented for visual indent

deftdawg committed:
Update wemo.py

trailing whitespace... (argh... the bot should just trim it)

deftdawg committed:
Back to you travis....

deftdawg committed:
Update wemo.py

deftdawg committed:
Update wemo.py

deftdawg committed:
Update wemo.py

kellerza committed:
Allow 4.5min startup time for recorder

deftdawg committed:
Update device_state_attributes only

This gets rid of the other stuff and just updates device_state_attributes, leaving the default properties alone.

pvizeli committed:
Refactory of envisalink (#6160)

* Refactory of envisalink

* remove event buss

* init dispatcher from hass.

* Move platform to new dispatcher

* fix lint

* add unittest & threadded functions

* fix copy & past error

deftdawg committed:
Incorporate comment suggestions

- Separate attribs from coffeemaker condition
- Set power units for threshold to mW to be consistent with others
- Adjust on-time labels to be more clear

kellerza committed:
Bugfix restore startup state (#6189)

balloob committed:
Remove automatically reloading group config (#6197)

lindsaymarkward committed:
Update Yeelight Sunflower light platform to 0.0.6 (#6208)

Add an optional extended description…

balloob committed:
Fix reporting on bad login (#6201)

molobrakos committed:
sensor.dovado: Upgraded library version

molobrakos committed:
sensor.eliqonline: Change icon

kellerza committed:
No wait for start and more async

balloob committed:
Version bump to 0.39

andrey-git committed:
Make glob preserve order (#6224)

fabaff committed:
Update regex (#6216)

balloob committed:
Merge pull request #6199 from home-assistant/release-0-39

0.39

balloob committed:
Update frontend

balloob committed:
Merge pull request #6227 from home-assistant/release-0-39

Release 0 39

balloob committed:
Fix recorder async (#6228)

balloob committed:
Merge pull request #6229 from home-assistant/release-0-39

Fix recorder async (#6228)

pvizeli committed:
Bugfix mqtt socket error (#6256)

balloob committed:
Config fix (#6261)

pvizeli committed:
Bugfix mqtt paho client to speend time (#6266)

balloob committed:
Version bump to 0.39.1

balloob committed:
Merge pull request #6272 from home-assistant/release-0-39-1

0.39.1

Commit:
Integration with lockitron

Commit:
Integration with lockitron

homeassistant commented:
Hello @aarya123,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`405805d323c34d6863fd14027711ce95cb5a4814`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant&#x2F;pull&#x2F;6306&#x2F;commits&#x2F;405805d323c34d6863fd14027711ce95cb5a4814) - This commit has something that looks like an email address (anubhaw.arya@hulu.com). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

Commit:
Line lengths

Commit:
Line lengths

homeassistant commented:
Hi @aarya123, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#6306).

Once you do that we will be able to review and accept this pull request.

Thanks!

Commit:
Another lint error

Commit:
Another lint error

homeassistant commented:
Hi @aarya123, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#6306).

Once you do that we will be able to review and accept this pull request.

Thanks!

Commit:
Line counter seems to be off

Commit:
Line counter seems to be off

Commit:
Another one

Commit:
Another one

zhelev committed:
Frontier silicon (#6131)

* added frontier_silicon constant

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* added frontier_silicon constant

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* changed info to debug

* added frontier_silicon constant

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* added the frontier_silicon component

* trying to satisfy pylint

* fsapi version 0.0.6

* remove white space

* generated requirements

* added the frontier_silicon component

* cleaning up according to travis

* trying to satisfy pylint

* trying to satisfy pylint

* fsapi version 0.0.6

* with fsapi version 0.0.7

* added fsapi dependency

* yielding the FSAPI

* Removing white space from docstring

* Removing white space from an empty line

* Switching to sync

* clean up white spaces and rename device to FSAPIDevice

* trying to satisfy pylint

* changed info to debug

* added the frontier_silicon component

* fsapi version 0.0.6

* generated requirements

* pylint

* moved import requests to the method where it is being used

* add a basic unit test

* cleaned up source code

* added frontier_silicon constant

* added the frontier_silicon component

* added basic test

* added fsapi to requirements_all.txt

* added coverage omit, though a basic test was included

* added MEDIA_TYPE_MUSIC for artist and album

* removed duplicate cons

* switched fsapi call to a property, removed unecessary comment

* detailed docstring for fs_device

* added a space for the info_name - info_text separator

* reduced proeprty (fsapi) access for volume down/up

emlove committed:
Version bump to 0.40.0.dev0

pvizeli committed:
Bootstrap / Component setup async (#6264)

* Bootstrap / Entiy setup async

* Cleanup add_job stuff / return task/future object

* Address paulus comments / part 1

* fix install pip

* Cleanup bootstrap / move config stuff to config.py

* Make demo async

* Further bootstrap improvement

* Address Martin's comments

* Fix initial tests

* Fix final tests

* Fix bug with prepare loader

* Remove no longer needed things

* Log error when invalid config

* More cleanup

* Cleanups platform events & fix lint

* Use a non blocking add_entities callback for platform

* Fix Autoamtion is setup befor entity is ready

* Better automation fix

* Address paulus comments

* Typo

* fix lint

* rename functions

* fix tests

* fix test

* change exceptions

* fix spell

e850205 committed:
Convert kpH and mpH to kph and mph (#6316)

There is no reason for the H to be capitalized. Changing it to lowercase increases consistency with other components and allows for use of the min/max sensor, which throws an error if the units of measurement are not the same.

deftdawg committed:
Rollback netdisco to 0.8.2 to resolve #6165 (#6314)

* Rollback netdisco to 0.8.2 to resolve #6165

* Rollback netdisco to 0.8.2 to resolve #6165

kellerza committed:
Log errors when loading yaml (#6257)

balloob committed:
Bootstrap tweaks tests (#6326)

* Update strings/fix component not found message.

* Fix tests

* More tweak text

molobrakos committed:
Merge pull request #6328 from molobrakos/eliqonline

sensor.eliqonline: Change to more appropriate icon

molobrakos committed:
Merge pull request #6329 from molobrakos/dovado

sensor.dovado: Upgraded library version

pvizeli commented on homeassistant/components/lock/lockitron.py:
```diff
@@ -0,0 +1,84 @@
+        self._state = self.do_change_request(SERVICE_UNLOCK)
+        self.update_ha_state()
+
+    def async_update(self):
```
Your platform is not async to use `async_update`. You can make this threadded with `update` or convert this to aiohttp for use with async context.

aarya123 replied:
Fixed!

scipioni committed:
Telegram webhooks new text event (#6301)

* new TELEGRAM_TEXT

* telegram command event renamed in 'telegram_command'

* fire telegram_text event anyway

pvizeli committed:
Cleanup component track_point_in_utc_time usage (#6330)

balloob committed:
Discovery fix (#6321)

* Fix incorrect import

* Create own discovery service

* Fix tests

* Fix hdmi_cec bad import

balloob committed:
Discovery fix (#6321)

* Fix incorrect import

* Create own discovery service

* Fix tests

* Fix hdmi_cec bad import

balloob committed:
Test against 3.6-dev (#6324)

pvizeli committed:
Bugfix ZigBee / Move from eventbus to dispatcher (#6333)

* Bugfix ZigBee / Move from eventbus to dispatcher

* fix lint

pvizeli committed:
Bugfix ZigBee / Move from eventbus to dispatcher (#6333)

* Bugfix ZigBee / Move from eventbus to dispatcher

* fix lint

balloob committed:
Bump netdisco to 0.9.1 (#6338)

balloob committed:
Bump netdisco to 0.9.1 (#6338)

balloob committed:
Version bump to 0.39.2

balloob committed:
Merge pull request #6339 from home-assistant/release-0-39-2

0.39.2

molobrakos committed:
sensor.dovado: compute state in update (#6340)

MartinHjelmare committed:
Fix mysensors callback race (#6311)

* Fix possible race at startup in mysensors callback

* Update devices via persistence before starting gateway to avoid
  two threads calling the same callback at the same time.

* Call add_devices max once per callback

fabaff committed:
Upgrade TwitterAPI to 2.4.5 (#6351)

fabaff committed:
Upgrade py-cpuinfo to 0.2.6 (#6335)

pvizeli committed:
Template sensor change flow / add restore (#6336)

andrey-git committed:
Zwave optimize value_added (#6210)

* Make zwave devices listen on less network changes.

* Convert more platforms

* Remove printouts.

* Fix copy-paste

* Change default dependent list to empty list

Commit:
Update Vagrant provision.sh (#6236)

- Bugfix: with f63a79ee we removed `script/home-assistant@.service`
  systemd unit file, which is used by Vagrant box to start/stop hass
- simplify interaction with Vagrant, provision.sh now is the only
  entry point and doesn't need the user to touch/remove files in
  order to change provisioner behavior

masarliev committed:
Update Adafruit_Python_DHT to support new raspberry kernel (#6325)

* Update Adafruit_Python_DHT to support new raspberry kernel

* update Adafruit Python DHT

reedriley committed:
Add fallback for name if userdevicename isn't set using old serialnumber logic (#6265)

Add an optional extended description…

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

titilambert committed:
Improve Honeywell US climate component (#5313)

* Improve Honeywell US climate component

* Fix tests

* Fix tests

* Add cool_away_temp and heat_away_temp for honeywell US

* Fix honeywell tests

* Fix PR comments

pvizeli committed:
Template binary_sensor change flow / add restore (#6343)

* Template binary_sensor change flow / add restore

* fix lint

Duoxilian committed:
Additional support for ecobee hold mode (#6258)

* Integrate suggestion in #5590 by nordlead2005. This change has been
sitting in limbo for over a month, but it is a good idea. I don't
mean to step on nordlead2005's toes, but we need to make progress.

* Use defined constant for TEMPERATURE_HOLD

* Integrate handling of vacation into hold mode. Canceling vacation
hold requires an update to the external pyecobee library. Creation
of vacation is not supported (it would be straightforward in the code,
but a complex user interface would be required, similar to what is
now done in the ecobee thermostat).

* Add capability to retrieve list of defined climates from ecobee.

* The mode() method used to return the system mode in internal
representation. However, the user sees a different notation in
the ecobee thermostat. Seeing some internal name is particularly
weired with user-defined climates, where these are named "smart1",
"smart2", etc., instead of the name the user has defined. Return
the user-defined name instead. This change might break some user
interfaces but is easily remedied (e.g., use "Away" instead of
"away").

* Simplify is_away_mode_on().

* Correction of erroneously indented else statement.

* Change comment as flake8 gets confused.

dramamoose committed:
Update Formulas in Convert XY to RGB (#6322)

* Update to Current RGB D65 Conversion

As per Philips Hue https://developers.meethue.com/documentation/color-conversions-rgb-xy

* Update the source of the XYZ to RGB formulas

* Fix Whitespace

* Update Whitespace

* Update Tests for new Formulas

* Update Tests

* Update XY_Brightness_to_hsv tests

* Update test_color.py

balloob committed:
Update coveragerc

emlove committed:
Use dynamic ports for test instances (#6232)

martinfrancois committed:
Added support for multiple codes executed in a row (#5908)

* Added support for multiple codes executed in a row

now codes can be specified either by simply providing a single code, which will then be sent like usual, or multiple codes can be executed in a row, specified in a comma delimited format in the configuration.yaml. For example: 111111,222222,333333,444444 would mean 111111 would be sent first, followed by 222222 and 333333 and 444444.

* rpi_rf: added line breaks to not exceed 79 characters per line

* include validation for correct formatting of codes

added regex which only allows either a single number (like 1252456245) or a sequence of commas followed by another number.

* added line breaks to not exceed 79 characters per line

* fix for 'continuation line under-indented for visual indent'

* another try at 'continuation line under-indented for visual indent'

* changed from regex to list for easier maintainability

* removed unnecessary splitting of strings

postlund committed:
Use push updates in Apple TV (#6323)

* Use push updates in Apple TV

* Fix review comments

jawilson committed:
Fix command sudo not found error in dev Dockerfile (#6346)

alanfischer committed:
Fix calendar authentication text, and handle calendar events without summaries. (#6337)

* Fixed google authorization text

* Let calendar handle events without a summary

pvizeli committed:
Move dispatcher out of init. (#6355)

andrey-git committed:
Zwave: Add remove/replace failed node services. (#6248)

* Zwave: Add remove/replace failed node services.

* Fix text

pvizeli committed:
Template switch change flow / add restore (#6356)

* Template switch change flow / add restore

* fix tests

* fix binary_sensor template

janLo committed:
Bump limitlessled dependency to 1.0.5. (#6334)

This fixes issue #6295.

GreenTurtwig committed:
Updated to catch timeout error

milaq committed:
snmp: upgrade pysnmp to 4.3.4 (#6359)

* snmp: upgrade pysnmp to 4.3.4

fixes https://github.com/home-assistant/home-assistant/issues/6238

* snmp: v4.3.4: add missing definition changes

pvizeli committed:
Bugfix new async_add_devices function (#6362)

pvizeli committed:
Restore for input_slider (#6360)

Commit:
Added IPv4 data collector (#6304)

* Added IPv4 data collector

* Formatting

* Bugfix: data is in kBit/s not kByte/s

jjmontesl committed:
Return None instead of raising ValueException from as_timestamp template function. (#6155)

kellerza committed:
[recorder] Catch more startup errors #6179 (#6192)

* [recorder] Catch more startup errors #6179

* Rebase on new recorder

happyleavesaoc committed:
twilio component (#6348)

* twilio component

* add http dependency to twilio

* fire->async_fire

andrey-git committed:
Add Z-Wave battery level as a sensor. (#6341)

pvizeli committed:
OwnTrack Async (#6363)

* Migrate owntrack to async

* fix tests

pvizeli committed:
Fix possibility that have multible topic subscribe mqtt (#6372)

pvizeli committed:
Migrate mqtt tracker and arwn sensor to async / cleanup owntrack (#6373)

* Migrate mqtt tracker and arwn sensor to async / cleanup owntrack

* Fix tests / lint

balloob committed:
Z-Wave prevent I/O event loop (#6369)

* Prevent Z-Wave I/O in event loop

* Move value_handler to util class.

* Add docstring

valentinalexeev committed:
Update pwaqi to 3.0 to use public API (#6376)

The underlying PWAQI library version 3.0 is now using public API to
access AQICN data.

mezz64 committed:
Update Hikvision Binary Sensors to latest library, remove pyDispatcher (#6231)

* Update pyHik version, remove pyDispatcher in favor of callbacks

* Fix naming

* Fix lint blank line

* Move stream thread start to HOMEASSISTANT_START event

* Bump library version to cleanup shutdown

* Fix requirements

colinodell committed:
Don't initialize components which have already been discovered (#6381)

* Don't initialize components which have already been discovered (fixes #5588)

* Don't log that we've found a service unless we know it's not a duplicate

* Encode discovery data hash with JSON

This also solves the issue of trying to hash non-hashable objects like dicts

* Add test for duplicate device discovery

colinodell committed:
Don't initialize components which have already been discovered (#6381)

* Don't initialize components which have already been discovered (fixes #5588)

* Don't log that we've found a service unless we know it's not a duplicate

* Encode discovery data hash with JSON

This also solves the issue of trying to hash non-hashable objects like dicts

* Add test for duplicate device discovery

balloob committed:
Version bump to 0.39.3

joe248 committed:
Comed Hourly Pricing sensor (#6378)

* Add ComEd RRTP price sensor

* Update wording to reflect ComEd's naming change from 'RRTP' to 'Hourly Pricing'

* Changed name of sensor source file

* Cleanup based on requested changes

* More cleanup

* small cleanups

Deinara committed:
Resolved issue #5688

Deinara committed:
tweaks

Deinara committed:
Changes for Travis bot. Unused variable 'on_ready' will likely remain reported

Deinara committed:
Edit docstring

balloob committed:
Merge pull request #6384 from home-assistant/release-0-39-3

0.39.3

titilambert committed:
Add multi contracts support for Hydroquebec (#6392)

andrey-git committed:
Add Zwave refresh services (#6377)

* Add Zwave refresh services

* services file

* Use dispatcher

* Add zwave prefix to signal

aronsky committed:
Add keep-alive feature to the generic thermostat (#6040)

* Add keep-alive feature to the generic thermostat

* Comply with maximum line lengths

* Added tests for the keep-alive functionality

siebert committed:
Fix wake_on_lan for german version of Windows 10 (#6397) (#6398)

Danielhiversen committed:
flux led lib (#6404)

jvolkman committed:
Small typo fix in setup_docker_prereqs

pvizeli committed:
Cleanup async handling (#6388)

* Cleanups unneeded blocks

* Cleanup bootstrap

* dedicated update_ha_state

* Fix imap_email_content

* fx tests

* Fix lint & spell

kellerza committed:
Restore for automation entities (#6254)

* Restore for automation entities

* coroutine

* no clue what i'm doing now

* Still passes nicely in py 3.4

Commit:
Use update instead of async_update

Commit:
Use update instead of async_update

balloob committed:
Fix tests no internet (#6411)

* Fix honeywell tests without internet

* Fix device tracker without internet

* Fix MFI using internet during tests

* Remove I/O from apns tests

balloob committed:
Prevent more I/O in apns (#6413)

pvizeli committed:
Restore flow on device_tracker platform (#6374)

* Restore flow on device_tracker platform

* fix flow

* fix lint

rytilahti committed:
switch.tplink: catch exceptions coming from pyHS100 to avoid flooding the logs when the plug is not available (#6400)

tchellomello committed:
Added sensors to support Ring.com devices (#6419)

titilambert committed:
Add Mint finance sensor (#6132)

* Add Mint finance sensor

* Add retry

* Fix PR comments

* Upgrade mintapi version

* Update mint_finance.py

* Doc tweak

* Update mint_finance.py

balloob committed:
Split bootstrap into bs + setup (#6416)

* Split bootstrap into bs + setup

* Lint

balloob committed:
Tweak recorder/restore_state (#6412)

* Tweak recorder/restore_state

* Lint

balloob committed:
Fix unnecessary warning for ip bans.yaml (#6417)

balloob committed:
Better restore_state warnings (#6418)

amelchio committed:
Set new color before turning LIFX bulbs on (#6402)

A LIFX bulb maintains its previous color even when the light is off.
For example, if the previous color is blue and the bulb is turned on
and then set to a red color, it will transition through purple colors.

After this commit, the target color is set while the bulb is still
turned off. This overrides the previous color and brightness that the
bulb remembered. The light is then turned on with the requested
transition duration.

For the example, this gives the expected result of only going through
red colors.

pvizeli commented on homeassistant/components/lock/lockitron.py:
```diff
@@ -0,0 +1,84 @@
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.lock import LockDevice
+from homeassistant.components.lock import PLATFORM_SCHEMA
```
Import this two things in one statement

pvizeli commented on homeassistant/components/lock/lockitron.py:
```diff
@@ -0,0 +1,84 @@
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.lock import LockDevice
+from homeassistant.components.lock import PLATFORM_SCHEMA
+from homeassistant.const import CONF_ACCESS_TOKEN, SERVICE_LOCK, \
```
Make you own constante for API usage. First SERVICE means service calls and second we can change this every time and you code will not work after that.

ishults committed:
Don't log username and password in camera url (#6390)

* Don't log username and password in camera url

* Attempt fix of tox issues

* Attempt to fix indentation issue

amelchio committed:
Ignore deleted mails in IMAP unread count (#6394) (#6395)

Message deletion in IMAP is a two step process: first delete, then expunge.
Deleting a message just sets a flag that usually makes the mail client hide
the message. It is the expunge that actually removes the message.

Thus, exclude the deleted messages so that the unread count matches up with
that of most mail clients.

andrey-git committed:
Delay zwave updates for 100ms to group them. (#6420)

* Add Zwave refresh services

* services file

* Use dispatcher

* Add zwave prefix to signal

* Delay zwave updates for 100ms to group them.

* Fixes

* lint

* Access _scheduled_update from loop thread only.

* More async

* Some optimizations

* Fix

andrey-git committed:
Rename _scheduled_update to _update_scheduled (#6434)

emlove committed:
Revert "Use dynamic port allocation for tests" (#6436)

Commit:
Tado device_tracker exception when mobile device has geofencing enabled but location is currently unknown. (#6401)

andrey-git committed:
Add a Z-wave workaround to do full refresh on update (#6403)

* Add Zwave refresh services

* services file

* Use dispatcher

* Add zwave prefix to signal

* Add a Z-wave workaround to do full refresh on update

dennisdegreef committed:
Use bundled certificates if port matches mqtts (#6429)

* Use bundled certificates if port matches mqtts

* Move import requests.certs to top, since it's used in more places

* Add happy and non-happy path tests for default certificate bundle on mqtts port

pvizeli committed:
Bugfix samsungtv discovery (#6438)

Commit:
Don't use external constants for API-specific constants

Commit:
Don't use external constants for API-specific constants

tchellomello committed:
Added unittest for Ring sensor (#6447)

Add an optional extended description…

balloob committed:
Shorten recorder connection init (#6432)

* Wait up to 9 seconds

* Set number of recorder retries to 8

* Do not sleep when reporting last connection error if no retries left

* Make sure we clean up old engine if connection is retrying

* Update __init__.py

bimbar committed:
KWB Easyfire support (#6018)

* KWB Easyfire Support

* requirements, coverage

* Initialization fun

* lint

* requirements bump

* lint

* Second best validation ...

* changes

* reworked validation

pvizeli commented:
Please do: `New files were added to .coveragerc.` and all other task in your init post.

Commit:
Bumped version number for supporting lib (#6462)

andersonshatch committed:
Send a logo with webostv notifications (#6380)

* Update to pylgtv 0.1.4

* Send icon with webostv notifications

Default to the homeassistant logo, but allow customizing it on the
component and for individual notifications

balloob committed:
Upgrade netdisco to 0.9.2 (#6466)

balloob committed:
Allow testing against uvloop (#6468)

appzer committed:
fix issue (#6470)

* fix issue

fix issue: https://community.home-assistant.io/t/error-in-new-notification-pushsafer/13308

* Update pushsafer.py

* Update requirements_all.txt

* Update pushsafer.py

* Update pushsafer.py

* Update pushsafer.py

aequitas committed:
Remove connection status state. (#6475)

Current implementation of connection status doesn't follow convention and is not properly configurable. Might be added again in the future as a full fledged entity or some other way.

For now users can rely on error logging to determine connection status.

fronzbot committed:
Support for Blink Camera System (#6444)

* Passing pep8, no tests yet

* Fixed some issues with the request throttling

* Removed ability to set throttle time because it was causing more issues than it was worth

* Added blink to .coveragerc

* Changed blinkpy version

* Removed global var, fixed per PR requests

* Added services for camera, migrated switch to binary_sensor

* Added schema for service, fixed naming, removed unused function

balloob committed:
Add warning for slow platforms/components (#6467)

* Add warning for slow platforms/components

* Add test for slow component setup.

* Add test for slow platform setup

* Fix tests on Py34

siebert committed:
Fix wake_on_lan ping for Linux. (#6480)

pvizeli committed:
Add dispatcher camera for internal image. (#6471)

* Add dispatcher camera for internal image.

* fix lint

* Add unittest

* Update dispatcher.py

pvizeli committed:
Add support for remove services / Reload script support (#6441)

* Add support for remove services / Reload script support

* Reload support for scripts

* Add more unittest for services

* Add unittest for script reload

* Address paulus comments

robbiet480 committed:
Expand MQTT lights (#6481)

* Add effect support to MQTT Light

* Use effect state topic for supported_features

* Dont use rainbow as default color

* Add color_temp support to MQTT JSON Light

* Add effect to MQTT JSON light

* Support lights in MQTT discovery

* Allow discovered devices to set their platform

* Add white value support to MQTT Light

* Add white value support to MQTT JSON Light

* Remove blank line

* Add color_temp support to MQTT Template light

* Add white value support to MQTT Template Light

* Remove unused SUPPORT_MQTT_TEMPLATE and stale unused flash and transition code from MQTT Template

* Add XY Color to MQTT Light Platform

* Fix syntax

* Fix more syntax errors

* Revert "Remove unused SUPPORT_MQTT_TEMPLATE and stale unused flash and transition code from MQTT Template"

This reverts commit c03798cb636e6b3258fce6f676fc6ca34c9b7d37.

* MQTT Template supports flash and transition but doesnt allow templating of the values

* Add XY color support to MQTT JSON

* Proper variable names

* Only allow whitelisted MQTT platforms to be loaded via MQTT Discovery

* Minor tweaks.

colinodell committed:
Allow configurable conditions for Pi-Hole sensor (#6465)

* Allow configurable conditions for Pi-Hole sensor

* Include all three conditions by default

* Share Pi-Hole API data across all sensors; eliminate redundant API calls

reedriley committed:
Improved iCloud 2FA support. (#5984)

pvizeli committed:
Android IP Cam support

arraylabs committed:
Update pymyq requirement (#6486)

* update pymyq to 0.0.7

* update pymyq requirement

Adds additional state for stopped door.

pvizeli committed:
add available

pvizeli committed:
fix lint

balloob committed:
Not always asume manufacturername is present (#6484)

emlove committed:
Catch exceptions on kodi commands

pvizeli committed:
Fix some things

pvizeli committed:
fix coverage

robbiet480 committed:
MQTT should load the provided platform not DOMAIN

See #6481

robbiet480 committed:
Add required changes to support MQTT discovery that I previously missed. See #6481.

pvizeli committed:
pump version 0.3 / make a lot of improvments

pvizeli committed:
fix lint

balloob committed:
Add first pass at Z-Wave light tests (#6483)

* Add first pass at Z-Wave light tests

* Remove unused SIGNAL_VALUE

* Lint

* Update test_init.py

pvizeli committed:
Fix lint

pvizeli committed:
Bugfix mqtt socket memory error (#6501)

* Bugfix mqtt socket memory error

* Fix tests

* Fix lint

amelchio committed:
Increase upper limit on light transitions (#2843)

This increases the global limit to 6535 seconds (1h48m55s) because that is
supported by all light platforms.

Some platforms support even longer transition times so the limit should
actually be platform specific.

happyleavesaoc committed:
more moon states

Deinara committed:
Added pylint ignore message and re-added .close()

Deinara committed:
Added an extra space

emlove committed:
Include functools.wraps

robbiet480 committed:
Merge pull request #6488 from pvizeli/pydroid-ipcam

Android IP Cam support

robbiet480 committed:
Merge pull request #6506 from happyleavesaoc/enhance-moon

more moon states

robbiet480 committed:
Merge pull request #6507 from amelchio/longer-light-transitions

Increase upper limit on light transitions (#2843)

robbiet480 committed:
Merge pull request #6494 from armills/kodi-command-exceptions

Catch exceptions on kodi commands

robbiet480 committed:
Make states constants

robbiet480 committed:
Merge pull request #6407 from jvolkman/patch-1

Small typo fix in setup_docker_prereqs

robbiet480 committed:
Merge pull request #6389 from Deinara/discord_connect_fix

Discord connect fix

robbiet480 committed:
Merge pull request #6361 from GreenTurtwig/dev

Updated to catch timeout error

robbiet480 committed:
Merge pull request #6001 from jumpkick/patch-2

Improvements for WeMo Insight switches

Commit:
Merge remote-tracking branch 'origin/master' into dev

pvizeli committed:
Bugfix android camera autodiscovery settings (#6510)

Add an optional extended description…

craigjmidwinter committed:
Insteon lib (#6505)

* use lib with caching to reduce collisions

* use 0.43

* change requirements

* update the  lib to 0.44

* update req

* fix typo

* just keep checking

* just keep checking - switch

* use 0.45 with file cache

* requirements

* Update requirements_all.txt

* Update insteon_local.py

* Update requirements_all.txt

* Update insteon_local.py

* update library

* fix lint

finish06 committed:
Update to Pyunifi2.0 (#6490)

* Updated pyUnifi

* Missing comma

* Security opt-out, not opt-in

* Adjust minimal values

* Update to pyUnifi 2.0

robbiet480 committed:
Don't allow sending to invalid iOS targets (#6115)

* Don't allow sending to invalid targets

* Fix valid target check

pvizeli committed:
Bugfix rpi_rf cleanup (#6513)

Add an optional extended description…

bachp committed:
[packaging] Include LICENSE.md in tarball (#6514)

pvizeli committed:
Android webcam better error handling / pump library 0.4 (#6518)

MartinHjelmare committed:
Fix mysensors gateway windows setup (#6500)

balloob committed:
Update frontend

balloob committed:
Remove mint finance sensor (#6522)

pavoni committed:
Append vera device id to entity id - but not name. (#6523)

* Append vera device id to entity id - but not name.

* Tidy.

* Tidy.

* Tidy after review.

* Re-order.

vrnagy committed:
Force update support for MQTT sensor (#6492)

Commit:
Wink scene(shortcut) support (#6147)

* Wink scene(shortcut) support

* Scenes to Scene

* Moved wink scenes from switches to scenes

* Updated python-wink version

bokub committed:
Add type configuration in history_stats (#6430)

pavoni committed:
Discovery is a dict rather than an array. (#6525)

amelchio committed:
Fix colortemp conversion for osramlightify (#6516)

* Fix colortemp conversion for osramlightify

Copied from the LIFX fix in 75df4be73391d5255371632f5fb08d7251e85b13.

* Fix style

* Updates from review

@armills:

While we're doing cleanup here, can you just change self._brightness,
self._rgb, self._name, self._temperature, and self._state assignments in
__init__ to None? These will get overwritten when self.update() is called, so
it's safer/cleaner to initialize them to None since it shouldn't matter if
everything is working.

emlove committed:
Update Kodi notifier to async (#6497)

* Update Kodi notifier to async

* Change Kodi CONF_SSL to CONF_PROXY_SSL

lwis committed:
Version bump to 0.41.0.dev0

balloob committed:
Simplify Android IP webcam discovery (#6528)

siebert committed:
Fix gen_requirements_all.py script for Windows. (#6547)

iamtpage committed:
Fix wake_on_lan ping with None as host (#6532)

* Update configuration validation

With the new update, wake_on_lan requires a host key in the configuration

* cast self._host to str as requested

* Changed host key back to optional

Commit:
Add to .coveragerc file

Commit:
Add to .coveragerc file

Commit:
Fix lint errors

Commit:
Fix lint errors

Commit:
Merge branch 'lockitron' of https://github.com/aarya123/home-assistant into lockitron

balloob commented:
Please rebase your branch on the latest dev. There are a lot of commits in here now that shouldn't be there.

pvizeli commented:
Please create a new PR on last dev branch.
